### DISCUSSION TITLE
Async & AWS SDK

### DISCUSSION THREAD

**Discussion by bmalinconico** [ioquatix/ioquatix#35] [Q&A] [answered]
Thank you for all your efforts in this area. I've been working on converting some critical paths to use your Async gem and got some nice speed and concurrency in some areas but I quickly ran into pile ups within the AWS SDK, which is tightly coupled to Net:HTTP.

My understanding is that Net::HTTP doesn't play nice with non-blocking I/O. If this assumption is incorrect please let me know.

I also reviewed [this gem](https://github.com/saluzafa/async-aws-ruby) which seems unmaintained. I also ran into issues with it.

Have you put any thought into how we could leverage async i/o with the aws sdk?

**Comment by ioquatix** [comment, marked as answer]
Net::HTTP should work just fine. What kind of problems are you running into?

**Reply by bmalinconico** [reply]
You are of course correct. I was seeing some flame graphs in DataDog which sent me down a hole and the there was conflicting information online about weather Net::HTTP played nice fibers.
